File: SozlukSwiftUI/SozlukSwiftUI/model/Entry.swift
//
//  Entry.swift
//  SozlukSwiftUI
//
//  Created by Alperen Saraç on 11.08.2025.
//

import Foundation
struct Entry: Codable {
    let id: Int
    let user_id: Int?
    let username: String?   // list endpoint’inde bazen "1" dönüyor, nullable tutalım
    let title: String
    let content: String
    let created_at: String
}

File: sozluk_flutter/lib/model/entry.dart
class Entry {
  final int id;
  final String title;
  final String content;
  final String createdAt;
  final String username;

  Entry({
    required this.id,
    required this.title,
    required this.content,
    required this.createdAt,
    required this.username,
  });

  factory Entry.fromJson(Map<String, dynamic> json) {
    return Entry(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      createdAt: json['created_at'],
      username: json['username'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'content': content,
      'created_at': createdAt,
      'username': username,
    };
  }
}
